### ISSUE TITLE
Undo problem with Preview Cleanup

### ISSUE THREAD

**Issue by WolfInABowl** [opentoonz/opentoonz#2491]
This is an issue where the Preview Cleanup mode seems to cause issues with undoing while it is active.
Steps:
1. Create a Raster level
2. Draw a line on the raster level
3. Enable Preview Cleanup mode
4. Press Ctrl-Z to undo - the line will disappear as if correctly undone
5. Disable Preview Cleanup mode - the line will reappear, having not actually been undone

**Comment by RodneyBaker** [comment]
I confirm this failure to Undo/maintain an Undo is still present in current v1.3 nightly releases.
At a guess I'd say this is a byproduct of all tools being disabled while preview is on except the Hand Tool, which is on by default when Cleanup Preview is activated.

Technically speaking I suppose Undo/Redo operations while Cleanup Preview is on should be disabled entirely.

**Comment by RodneyBaker** [comment]
This report has me curious of the potential benefits in using Undo/Redo (or history) in the cleanup process.

@WolfInABowl Can you share any additional insight into the workflow of cleanup and how undo/redo would ideally work in the process?

I do note that if cleanup is merely interfering with the standard history then a reasonable way forward would be to exclude it from cleanup (which has other restoration processes) but I am curious  what success might look like if history was fully implemented and even improved the cleanup process.

In short, your feedback will help in determining the best way forward.

**Comment by RodneyBaker** [comment]
Closing as likely resolved in current nightly releases.
@WolfInABowl Please let us know if you are still experiencing this problem.
